src/textual/_xterm_parser.py

```diff
@@ -108,21 +108,29 @@ def parse(self, on_token: TokenCallback) -> Generator[Awaitable, str, None]:
                 while True:
                     sequence += yield read1()
                     self.debug_log(f"sequence={sequence!r}")
-                    keys = get_ansi_sequence(sequence, None)
+                    # Was it a pressed key event that we received?
+                    keys = get_key_ansi_sequence(sequence, None)
                     if keys is not None:
                         for key in keys:
                             on_token(events.Key(self.sender, key=key))
                         break
-                    else:
-                        mouse_match = _re_mouse_event.match(sequence)
-                        if mouse_match is not None:
-                            mouse_code = mouse_match.group(0)
-                            event = self.parse_mouse_code(mouse_code, self.sender)
-                            if event:
-                                on_token(event)
-                            break
+                    # Or a mouse event?
```

Comment:
I removed the `else` level of indentation here, because keeping it would have introduced the need to also add a new level of indent to manage the "mode reports" and that would have been quite ugly :sweat_smile: 
The `break` statements here acts as _"early returns"_ so we should be good without this `else:` level :slightly_smiling_face: